packages/core/src/node/utils/detectReactVersion.ts

```diff
@@ -65,5 +65,14 @@ export async function resolveReactAlias(reactVersion: number) {
       }
     }),
   );
+
+  // react-dom/server should resolve to browser when targets is web
+  if (alias?.['react-dom/server'] && !isSSR) {
```

Comment:
Can we change the resolve options of `ResolverFactory.createResolver` to resolve the browser bundles correctly?

Reply:
done，this logic only handles react-related dependencies, so modifying `conditioalNames` should have no impact.